foundations/src/telemetry/settings/server.rs

```diff
@@ -16,15 +13,15 @@ pub struct TelemetryServerSettings {
 
     /// Telemetry server address.
     #[serde(default = "TelemetryServerSettings::default_addr")]
-    pub addr: SocketAddr,
+    pub addr: ListenAddr,
```

Comment:
This is still a breaking change, but we should release foundations 5 anyway for the hyper 1 switch.